solution/index.js

```diff
@@ -0,0 +1,306 @@
+      })
+    }
+  }
+}
+
+function addOrDeleteTask(event) {
+  // one click event works on all three buttons and only on buttons
+  const button = event.target
+  if (button.className === 'remove-task') taskRemover(button)
+
+  if (button.className !== 'bottom-btn') return
+  // The rest is for adding a task
+
+  // gets the section and input that are relavent to the button
+  const section = button.closest('section')
+  const input = document.querySelector(
+    `#${section.id} > .bottom-of-section > .bottom-input`
+  )
+  
+  // I don't allow the user to enter an empt input
+  if (!input.value) {
+    alert(`You can't submit an empty task`)
+    return
+  }
+
+  // I select the list within the section and append a list element
+  const list = document.querySelector(`#${section.id} > ul`)
+  list.prepend(createListElement('li', input.value, ['task'], {}))
+  
+  // I update the local storage
+  listStorage[section.id].unshift(input.value)
+  localStorage.setItem('tasks', JSON.stringify(listStorage))
+  
+  // I clear the input
+  input.value = ''
+  
+  // I focus on the input field so the user can continuously type more tasks
+  input.focus()
+}
+
+function taskRemover(remover){
+  const taskRemoved = remover.closest('.task')
+  
+  // update Local Storage
+  const positionInList = [...taskRemoved.parentNode.children].indexOf(taskRemoved)
+  listStorage[taskRemoved.closest('section').id].splice(positionInList, 1)
+  localStorage.setItem('tasks', JSON.stringify(listStorage))
+  
+  taskRemoved.remove()
+  
+}
+
+function editTask(event) {
+  // function only effects list elements
+  const listEl = event.target
+  if (listEl.tagName !== 'LI') return
+  
+  // I get the list text without including my task remover
+  const listElText = listEl.textContent
+  
+  // I switch the list element with a input
+  const temporaryInput = createElem('input', [listElText], 'task')
+  listEl.parentNode.replaceChild(temporaryInput, listEl)
+  temporaryInput.focus()
+  
+  // When I leave the input it changes back to the list Element
+  temporaryInput.onblur = () => {
+    // update the text content of the list element (can't be left empty)
+    listEl.textContent = temporaryInput.value
+    ? temporaryInput.value
+      : listEl.textContent
+      
+    temporaryInput.parentNode.replaceChild(listEl, temporaryInput)
+
+    // update the local storage
+    // I find the index of listEl in the Dom, and use it to find where it is in my listStorage
+    const positionInList = [...listEl.parentNode.children].indexOf(listEl)
+    listStorage[listEl.closest('section').id][positionInList] =
+    listEl.textContent
+    localStorage.setItem('tasks', JSON.stringify(listStorage))
+  }
+}
+
+// This function saves the list element I'm hovering over to the DOM
+function listELHoveringOver(event) {
+  if (!event.target.closest('.task')) return
+  tasksContainer.hoveringOver = event.target
+  
+  event.target.onmouseout = () => {
+    tasksContainer.hoveringOver = null
```

Comment:
That's basically a global variable in disguise. Try to think of a better way to pass the hovered element into the keyboard event handler.